gateway/mw_oas_validate_request.go

```diff
@@ -52,7 +53,15 @@ func (k *ValidateRequest) ProcessRequest(w http.ResponseWriter, r *http.Request,
 		return nil, http.StatusOK
 	}
 
-	router, err := gorillamux.NewRouter(&k.Spec.OAS.T)
+	oasSpec := k.Spec.OAS.T
+	apiURL := getAPIURL(*k.Spec.APIDefinition, k.Gw.GetConfig())
+	u, err := url.Parse(apiURL)
```

Comment:
This err needs a check

Reply:
changed this part to be simpler, keeping relative URL to the listen path will solve the problem